wagtail/admin/templates/wagtailadmin/shared/header.html

```diff
@@ -8,19 +8,24 @@
     - `search_url` - if present, display a search box. This is a URL route name (taking no parameters) to be used as the action for that search box
     - `query_parameters` - a query string (without the '?') to be placed after the search URL
     - `icon` - name of an icon to place against the title
+    - `avatar` - if present, display an 'avatar' in place of icon. This is the URL to be used as the img src for avatar 
     - `merged` - if true, add the classname 'merged'
     - `action_url` - if present, display an 'action' button. This is the URL to be used as the link URL for the button
     - `action_text` - text for the 'action' button
     - `action_icon` - icon for the 'action' button, default is 'icon-plus'
 
 {% endcomment %}
-<header class="header {% if merged %}merged{% endif %} {% if search_form %}hasform{% endif %}">
+<header class="header {{ classname }} {{ merged|yesno:" merged ," }}{{ search_form|yesno:" hasform ," }}">
     {% block breadcrumb %}{% endblock %}
     <div class="row">
         <div class="left">
             <div class="col">
                 <h1 class="header__title">
-                    {% if icon %}{% icon name=icon %}{% endif %}
+                    {% if icon %}
+                        {% icon class_name="w-header__icon" name=icon %}
+                    {% elif avatar %}
+                        <div class="avatar"><img src="{{ avatar }}" alt="" /></div>
+                    {% endif %}
```

Comment:
This is not really positioned correctly yet, so will need something like 

https://github.com/wagtail/wagtail/blob/ffda5580159094c941053a78fbaea1e1980823fb/client/scss/components/_header.scss#L21-L23

For the `h1 .avatar`

Reply:
another approach may be a neg margin.
`margin-inline-start: -2rem;`
(not sure on amount, just guessing)